src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-config-modal/rgw-config-modal.component.ts

```diff
@@ -0,0 +1,141 @@
+    private router: Router,
+    public actionLabels: ActionLabelsI18n,
+    private rgwBucketService: RgwBucketService,
+    private rgwEncryptionModal: RgwBucketEncryptionModel,
+    private notificationService: NotificationService
+  ) {
+    this.createForm();
+  }
+  ngOnInit(): void {
+    this.kmsProviders = this.rgwEncryptionModal.kmsProviders;
+    this.authMethods = this.rgwEncryptionModal.authMethods;
+    this.secretEngines = this.rgwEncryptionModal.secretEngines;
+  }
+
+  createForm() {
+    this.configForm = this.formBuilder.group({
+      address: [
+        null,
+        [
+          Validators.required,
+          CdValidators.custom('vaultPattern', (value: string) => {
+            if (_.isEmpty(value)) {
+              return false;
+            }
+            return !this.vaultAddress.test(value);
+          })
+        ]
+      ],
+      kms_provider: ['vault', Validators.required],
+      encryptionType: [null, Validators.required],
+      auth_method: [null, Validators.required],
+      secret_engine: [null, Validators.required],
+      secret_path: ['/'],
+      namespace: [null],
+      token: [
+        null,
+        [
+          CdValidators.requiredIf({
+            auth_method: 'token'
+          })
+        ]
+      ],
+      ssl_cert: [null, CdValidators.sslCert()],
+      client_cert: [null, CdValidators.pemCert()],
+      client_key: [null, CdValidators.sslPrivKey()],
+      role: [
+        null,
+        [
+          CdValidators.requiredIf({
+            auth_method: 'agent'
+          })
+        ]
+      ],
+      kmsEnabled: [{ value: false }],
+      s3Enabled: [{ value: false }]
+    });
+  }
+
+  fileUpload(files: FileList, controlName: string) {
+    const file: File = files[0];
+    const reader = new FileReader();
+    reader.addEventListener('load', () => {
+      const control: AbstractControl = this.configForm.get(controlName);
+      control.setValue(file);
+      control.markAsDirty();
+      control.markAsTouched();
+      control.updateValueAndValidity();
+    });
+  }
+
+  onSubmit() {
+    const values = this.configForm.value;
+    this.rgwBucketService
+      .setEncryptionConfig(
+        values['encryptionType'],
+        values['kms_provider'],
+        values['auth_method'],
+        values['secret_engine'],
+        values['secret_path'],
+        values['namespace'],
+        values['address'],
+        values['token'],
+        values['owner'],
+        values['ssl_cert'],
+        values['client_cert'],
+        values['client_key']
+      )
+      .subscribe(
+        () => {
+          this.notificationService.show(
+            NotificationType.success,
+            $localize`Updated RGW Encryption Configuration values`
+          );
+          this.activeModal.close();
+          this.router.routeReuseStrategy.shouldReuseRoute = () => false;
+          this.router.onSameUrlNavigation = 'reload';
+          this.router.navigate([this.router.url]);
```

Comment:
I believe the modalClose and routing should be moved to the `complete:` block of subscribe. Also, one issue I am seeing with the reload is that, it kind of resets the form. So I think its good to show a modal with a warning saying that the bucket form will be reset once you proceed. WDYT?

Reply:
At least we can create a tracker for the latter.